File: templates/kmp/screens/settings/view/SettingsScreen.kt
package features.settings.view

import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import core.theme.AppColors

@Composable
fun SettingsScreen(
    onNavigate: (String) -> Unit = {},
) {
    var notifications by remember { mutableStateOf(true) }
    var darkMode by remember { mutableStateOf(true) }
    var biometrics by remember { mutableStateOf(false) }

    LazyColumn(modifier = Modifier.fillMaxSize().padding(16.dp)) {
        item {
            Text("Settings", style = MaterialTheme.typography.headlineLarge, color = AppColors.TextPrimary)
            Spacer(Modifier.height(24.dp))
            Text("PREFERENCES", style = MaterialTheme.typography.bodySmall, color = AppColors.TextMuted)
            Spacer(Modifier.height(8.dp))
        }
        item { ToggleRow("🔔", "Push Notifications", notifications) { notifications = it } }
        item { ToggleRow("🌙", "Dark Mode", darkMode) { darkMode = it } }
        item { ToggleRow("🔒", "Biometric Login", biometrics) { biometrics = it } }
        item {
            Spacer(Modifier.height(24.dp))
            Text("ACCOUNT", style = MaterialTheme.typography.bodySmall, color = AppColors.TextMuted)
            Spacer(Modifier.height(8.dp))
        }
        item { NavRow("🔑", "Change Password") { onNavigate("change-password") } }
        item { NavRow("🛡️", "Privacy Policy") { onNavigate("privacy-policy") } }
        item { NavRow("📄", "Terms of Service") { onNavigate("terms") } }
        item { NavRow("❓", "FAQ") { onNavigate("faq") } }
        item { NavRow("ℹ️", "About") { onNavigate("about") } }
    }
}

@Composable
private fun ToggleRow(icon: String, title: String, value: Boolean, onChanged: (Boolean) -> Unit) {
    Row(modifier = Modifier.fillMaxWidth().padding(vertical = 8.dp), horizontalArrangement = Arrangement.SpaceBetween) {
        Row { Text("$icon  ", style = MaterialTheme.typography.titleMedium); Text(title, color = AppColors.TextPrimary) }
        Switch(checked = value, onCheckedChange = onChanged, colors = SwitchDefaults.colors(checkedThumbColor = AppColors.Primary))
    }
}

@Composable
private fun NavRow(icon: String, title: String, onClick: () -> Unit) {
    Row(
        modifier = Modifier.fillMaxWidth().clickable(onClick = onClick).padding(vertical = 12.dp),
        horizontalArrangement = Arrangement.SpaceBetween,
    ) {
        Row { Text("$icon  ", style = MaterialTheme.typography.titleMedium); Text(title, color = AppColors.TextPrimary) }
        Text("›", style = MaterialTheme.typography.headlineSmall, color = AppColors.TextMuted)
    }
}

File: templates/flutter/screens/settings/presentation/settings_screen.dart
import 'package:flutter/material.dart';
import 'package:go_router/go_router.dart';
import '../../../core/theme/app_colors.dart';

class SettingsScreen extends StatefulWidget {
  const SettingsScreen({super.key});
  @override
  State<SettingsScreen> createState() => _SettingsScreenState();
}

class _SettingsScreenState extends State<SettingsScreen> {
  bool _notifications = true;
  bool _darkMode = true;
  bool _biometrics = false;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: const Text('Settings')),
      body: ListView(
        padding: const EdgeInsets.all(16),
        children: [
          _sectionTitle('Preferences'),
          _toggleTile('Push Notifications', '🔔', _notifications, (v) => setState(() => _notifications = v)),
          _toggleTile('Dark Mode', '🌙', _darkMode, (v) => setState(() => _darkMode = v)),
          _toggleTile('Biometric Login', '🔒', _biometrics, (v) => setState(() => _biometrics = v)),
          const SizedBox(height: 24),
          _sectionTitle('Account'),
          _navTile('Change Password', '🔑', () => context.push('/change-password')),
          _navTile('Privacy Policy', '🛡️', () => context.push('/privacy-policy')),
          _navTile('Terms of Service', '📄', () => context.push('/terms')),
          _navTile('FAQ', '❓', () => context.push('/faq')),
          _navTile('About', 'ℹ️', () => context.push('/about')),
        ],
      ),
    );
  }

  Widget _sectionTitle(String title) => Padding(
    padding: const EdgeInsets.only(bottom: 8, left: 4),
    child: Text(title, style: Theme.of(context).textTheme.bodySmall?.copyWith(letterSpacing: 1, fontWeight: FontWeight.w600)),
  );

  Widget _toggleTile(String title, String icon, bool value, ValueChanged<bool> onChanged) => ListTile(
    leading: Text(icon, style: const TextStyle(fontSize: 20)),
    title: Text(title),
    trailing: Switch(value: value, onChanged: onChanged, activeColor: AppColors.primary),
    contentPadding: const EdgeInsets.symmetric(horizontal: 8),
  );

  Widget _navTile(String title, String icon, VoidCallback onTap) => ListTile(
    leading: Text(icon, style: const TextStyle(fontSize: 20)),
    title: Text(title),
    trailing: const Icon(Icons.chevron_right, color: AppColors.textMuted),
    onTap: onTap,
    contentPadding: const EdgeInsets.symmetric(horizontal: 8),
  );
}
